File: java/96A.java
// 96 A. Football

import java.util.Scanner;

class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String s = sc.next();

        int count = 1;
        boolean dangerous = false;

        for(int i=1; i<s.length(); ++i) {
            if(s.charAt(i) == s.charAt(i-1)) {
                count++;
                if(count >= 7) {
                    dangerous = true;
                    break;
                }
            } else {
                count = 1;
            }
        }

        System.out.println(dangerous ? "YES" : "NO");
        sc.close();
    }
}

File: cpp/96A.cpp
// 96 A. Football

#include <iostream>
#include <string>
using namespace std;

int main() {
    string s;
    cin >> s;

    int count = 1;
    bool dangerous = false;

    for (int i = 1; i < (int)s.length(); i++) {
        if (s[i] == s[i-1]) {
            count++;
            if (count >= 7) {
                dangerous = true;
                break;
            }
        } else {
            count = 1;
        }
    }

    cout << (dangerous ? "YES" : "NO") << endl;
    return 0;
}

